src/Core/Import/File/CsvFileReader.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Import\File;
+
+use PrestaShop\PrestaShop\Core\Import\File\DataRow\DataRow;
+use SplFileInfo;
+
+/**
+ * Class CsvFileReader defines a CSV file reader.
+ */
+final class CsvFileReader implements FileReaderInterface
+{
+    /**
+     * @var string the data separator in the CSV row
+     */
+    private $separator;
+
+    /**
+     * @param string $separator
+     */
+    public function __construct($separator)
+    {
+        $this->separator = $separator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function read(SplFileInfo $file)
+    {
+        $handle = fopen($file->getPathname(), 'r');
+
+        while ($row = fgetcsv($handle, 0, $this->separator)) {
+            yield DataRow::createFromArray($row);
```

Comment:
I believe `yield` existed in php5 but I'm not 100% sure. Can you confirm this is php5.6-compliant ?

Reply:
It was introduced in php 5.5 🙂 http://php.net/manual/en/migration55.new-features.php